.pre-commit-config.yaml

```diff
@@ -11,13 +11,26 @@ repos:
       - id: prettier
         additional_dependencies:
           - prettier@3.5.0
-
-  - repo: https://github.com/editorconfig-checker/editorconfig-checker.python
-    rev: "3.2.0"
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v5.0.0
     hooks:
-      - id: editorconfig-checker
-        alias: ec
-
+      - id: trailing-whitespace
+        args: [--markdown-linebreak-ext=md]
+        exclude: |
+          (?x)^(
+              .*\.snap$|
+              nf_core/pipeline-template/subworkflows/.*|
+              nf_core/pipeline-template/modules/.*|
+              tests/pipelines/__snapshots__/.*
+          )$
+      - id: end-of-file-fixer
+        exclude: |
+          (?x)^(
+              .*\.snap$|
+              nf_core/pipeline-template/subworkflows/.*|
+              nf_core/pipeline-template/modules/.*|
+              tests/pipelines/__snapshots__/.*
```

Comment:
looks like we need to also exclude main.nf in modules and subworkflow template files

Reply:
I think modifying the template is good, no? What we should skip are the modules and subworkflows because they are handeled in the module repo

Reply:
i am just wondering about the introduced changes... how come they were not there with editorconfig?

Reply:
I can't see anywhere where we excluded these files... but I have tested creating a new pipeline and running pre-commit there and everything passes.